src/store/postSlice.js

```diff
@@ -17,6 +23,57 @@ export const getAllPosts = createAsyncThunk("posts/getPosts", async () => {
   }
 });
 
+export const bookmarkPosts = createAsyncThunk(
+  "posts/bookmarkPosts",
+  async ({ postId, token }) => {
+    try {
+      const response = await bookmarkService(postId, token);
+      console.log(response);
+      return response.data.bookmarks;
+    } catch (error) {
+      console.error(error);
```

Comment:
Remove the console.log if not needed. In case of error, you can display the error to the UI itself